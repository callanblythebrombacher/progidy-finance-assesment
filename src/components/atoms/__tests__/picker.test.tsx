import React from 'react';
import {shallow} from 'enzyme';
import DropDownPicker from 'react-native-dropdown-picker';
import {PickerComponent} from '../picker.atom.ts';

import {describe, jest, beforeEach, afterEach, it, expect} from '@jest/globals';

describe('PickerComponent Atom', () => {
  const mockItems = [
    {label: 'Option 1', value: 1},
    {label: 'Option 2', value: 2},
    {label: 'Option 3', value: 3},
  ];

  const mockSetValue = jest.fn();
  const mockValue = 1;

  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <PickerComponent
        items={mockItems}
        setValue={mockSetValue}
        value={mockValue}
      />,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('passes correct props to DropDownPicker', () => {
    const dropDownPicker = wrapper.find(DropDownPicker);
    expect(dropDownPicker).toHaveLength(1);
    expect(dropDownPicker.prop('open')).toBe(false); // Initial state of 'open'
    expect(dropDownPicker.prop('value')).toBe(mockValue);
    expect(dropDownPicker.prop('items')).toEqual(mockItems);
    expect(typeof dropDownPicker.prop('setOpen')).toBe('function');
    expect(typeof dropDownPicker.prop('setValue')).toBe('function');
    expect(typeof dropDownPicker.prop('setItems')).toBe('function');
  });

  it('triggers setValue function on value change', () => {
    const dropDownPicker = wrapper.find(DropDownPicker);
    dropDownPicker.simulate('change', 'option2');
    expect(mockSetValue).toHaveBeenCalledWith('option2');
  });

  it('toggles open state when DropDownPicker is clicked', () => {
    const dropDownPicker = wrapper.find(DropDownPicker);
    dropDownPicker.simulate('open');
    expect(wrapper.find(DropDownPicker).prop('open')).toBe(true);
    dropDownPicker.simulate('close');
    expect(wrapper.find(DropDownPicker).prop('open')).toBe(false);
  });
});
