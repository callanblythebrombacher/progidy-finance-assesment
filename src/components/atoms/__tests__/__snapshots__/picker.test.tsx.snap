// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerComponent Atom renders correctly 1`] = `
<View
  style={
    {
      "marginLeft": 10,
      "width": 140,
      "zIndex": 3,
    }
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "zIndex": 5000,
        },
        {},
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderColor": "#000",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minHeight": 50,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 3,
          "width": "100%",
          "zIndex": 5000,
        }
      }
      testID="dropdown-picker"
    >
      <Text
        style={
          [
            {
              "color": "#000",
              "flex": 1,
            },
            {},
            {},
            false,
          ]
        }
      >
        Option 1
      </Text>
      <View
        style={
          [
            {
              "marginLeft": 10,
            },
            {},
          ]
        }
      >
        <Image
          source={
            {
              "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
            }
          }
          style={
            [
              {
                "height": 20,
                "width": 20,
              },
              {},
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
